direct/nn/openvino/openvino_model.py

```diff
@@ -0,0 +1,107 @@
+from openvino_extensions import get_extensions_path
+from openvino.inference_engine import IECore
+
+import torch
+import torch.nn as nn
+import io
+
+
+class InstanceNorm2dFunc(torch.autograd.Function):
+    @staticmethod
+    def symbolic(g, cls, input):
+        c_scale = g.op("Constant", value_t=cls.scale_one)
+        c_bias = g.op("Constant", value_t=cls.bias_zero)
+        return g.op("InstanceNormalization", input, c_scale, c_bias)
+
+    @staticmethod
+    def forward(self, cls, input):
+        return cls.origin_forward(input)
+
+
+class InstanceNorm2dONNX(nn.InstanceNorm2d):
+    """
+    This is a support class which helps export network with InstanceNorm2d in ONNX format.
+    """
+
+    def __init__(self, num_features):
+        super().__init__(num_features)
+        self.origin_forward = super().forward
+        self.scale_one = torch.ones(num_features)
+        self.bias_zero = torch.zeros(num_features)
+
+    def forward(self, input):
+        return InstanceNorm2dFunc.apply(self, input).clone()
+
+
+def convert_layer(model):
+    for name, l in model.named_children():
+        layer_type = l.__class__.__name__
+        if layer_type == "InstanceNorm2d":
+            new_layer = InstanceNorm2dONNX(l.num_features)
+            setattr(model, name, new_layer)
+        else:
+            convert_layer(l)
+
+
+class OpenVINOModel(nn.Module):
+    def __init__(self, model):
+        super().__init__()
+        self.inputs: tuple
+        self.input_names: list
+        self.model = model
+        self.exec_net = None
+        self.model_name = self.model.__class__.__name__
+
+    def create_net(self, *args, **kwargs):
+        self.inputs = args
+
+        if self.model_name == "RIM":
+            self.input_names = ["input_image", "masked_kspace", "sampling_mask", "sensitivity_map"]
+            output_names = ["cell_outputs", "previous_state"]
+        elif self.model_name == "Unet2d":
+            output_names = ["output"]
+            self.input_names = ["masked_kspace"]
+            if self.model.image_initialization == "sense":
+                self.input_names.append("sensitivity_map")
+        else:
+            raise ValueError(f"The model is not supported by OpenVINO: {self.model_name}")
+
+        for k in self.input_names:
+            if k in kwargs:
+                self.inputs += tuple([kwargs[k]])
+
+        ie = IECore()
+        ie.add_extension(get_extensions_path(), "CPU")
+
+        convert_layer(self.model)
+
+        with torch.no_grad():
```

Comment:
Do we need `torch.no_grad()` here?

Reply:
Yes, otherwise some of the tests will fail